### PR TITLE
[manta] Dziabko/dolphin ci workflows

### LINKED ISSUE #376: CI workflows for Dolphin runtime
We need a benchmarking workflow to generate custom weights.
Maybe add to the draft-release workflow as well down the line.

### PR DESCRIPTION
**Pull request by dziabko**
## Description

Adds a new dolphin runtime CI workflow

closes: #376 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (`manta` or `dolphin`) with right title (start with [Manta] or [Dolphin]),
- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests.
- [ ] Updated relevant documentation in the code.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. If this number is updated, then the spec_version must also be updated 
- [ ] If needed, notify the committer this is a draft-release and a tag is needed after merging the PR.
- [ ] Verify benchmarks & weights have been updated for any modified runtime logics
- [ ] If needed, bump `version` for every crate.
- [ ] If import a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- [ ] If needed, update our Javascript/Typescript APIs. These APIs are offcially used by exchanges or community developers.
- [ ] If we're going to issue a new release, freeze the code one week early(it depends, but usually it's one week), ensure we have enough time for related testing.

### CODE REVIEW

File: .github/workflows/generate_dolphin_weights_files.yml [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+              name: pallet_treasury
+            iterations: 20
+          -
+            extrinsic:
+              id: '*'
+              name: manta_collator_selection
+            pallet:
+              id: manta_collator_selection
+              name: manta_collator_selection
+            iterations: 20
```
Comment (ghzlatarev) [inline .github/workflows/generate_dolphin_weights_files.yml:210]: Could you check if we can make the matrix dynamic and pass a list of the pallets you want to benchmark as input.
If there's no input we can have this as default. Kind of out of scope, but we'd be able to dedupe the 3 workflows if we can do that.

**Comment by stechu** [comment]
@dziabko  Any follow up on this PR?

File: .github/workflows/generate_dolphin_weights_files.yml [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    needs:
+      - start-node-builder-current
+      - build-benchmark
+    runs-on: ${{ needs.start-node-builder-current.outputs.runner-label }}
+    strategy:
+      matrix:
+        benchmark:
+          extrinsic:
+            id: '*'
+            name: [frame_system, pallet_balances, pallet_collective, pallet_democracy, pallet_membership, pallet_multisig, pallet_scheduler, pallet_session, pallet_tx_pause, pallet_utility, pallet_timestamp, pallet_treasury, manta_collator_selection]
```
Comment (ghzlatarev) [inline .github/workflows/generate_dolphin_weights_files.yml]: These are the pallet names not the extrinsics.

File: .github/workflows/generate_dolphin_weights_files.yml (hunk 1 above)
Comment (ghzlatarev) [reply .github/workflows/generate_dolphin_weights_files.yml:210]: Nvm this won't work

File: .github/workflows/publish-draft-releases.yml [hunk 3]
```diff
@@ -1129,6 +1133,8 @@ jobs:
             name: calamari
           -
             name: manta
+          -
+            name: dolphin
```
Comment (ghzlatarev) [inline .github/workflows/publish-draft-releases.yml:1137]: You need to define some more output variables for the dolphin runtime like `calamari-runtime-current`, `manta-runtime-current`, etc See how they are used and do the same thing. Basically the runtimes version must be equal or the job fails.

**Pull request closed without merging**